controllers/nutanixmachine_controller.go

```diff
@@ -232,12 +233,22 @@ func (r *NutanixMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	v3Client, err := CreateNutanixClient(ctx, r.SecretInformer, r.ConfigMapInformer, ntxCluster)
+	v3Client, err := prismclientcache.DefaultCache.Get(ntxCluster.Name)
 	if err != nil {
-		conditions.MarkFalse(ntxMachine, infrav1.PrismCentralClientCondition, infrav1.PrismCentralClientInitializationFailed, capiv1.ConditionSeverityError, err.Error())
-		return ctrl.Result{Requeue: true}, fmt.Errorf("client auth error: %v", err)
+		if errors.Is(err, prismclientcache.ErrorClientNotFound) {
+			log.Info("Nutanix client not found in cache. Creating new client")
```

Comment:
nit: Nutanix client not found in cache. => Nutanix pc client is not found in cache.